packages/cozy-stack-client/src/OAuthClient.js

```diff
@@ -0,0 +1,358 @@
+
+    return this.fetch(
+      'DELETE',
+      `/auth/register/${this.oauthOptions.clientID}`,
+      null,
+      {
+        credentials: this.registrationAccessTokenToAuthHeader()
+      }
+    )
+  }
+
+  /**
+   * Fetches the complete set of client informations from the server after it has been registered.
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @returns {promise}
+   */
+  async fetchInformations() {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+
+    return this.fetch(
+      'GET',
+      `/auth/register/${this.oauthOptions.clientID}`,
+      null,
+      {
+        credentials: this.registrationAccessTokenToAuthHeader()
+      }
+    )
+  }
+
+  /**
+   * Updates the client own information. This method will update both the local information and the remote information on the OAuth server.
+   * @throws {NotRegisteredException} When the client doesn't have it's registration information
+   * @param   {object} informations Set of informations to update. Note that some fields such as `clientID` can't be updated.
+   * @param   {boolean} resetSecret = false Optionnal, whether to reset the client secret or not
+   * @returns {promise} A promise that resolves to a complete, updated list of client informations
+   */
+  async updateInformations(informations, resetSecret = false) {
+    if (!this.isRegistered()) throw new NotRegisteredException()
+
+    const mandatoryFields = {
+      clientID: this.oauthOptions.clientID,
+      clientName: this.oauthOptions.clientName,
+      redirectURI: this.oauthOptions.redirectURI,
+      softwareID: this.oauthOptions.softwareID
+    }
+    const data = this.snakeCaseOAuthData({
+      ...mandatoryFields,
+      ...informations
+    })
+
+    if (resetSecret) data['client_secret'] = this.oauthOptions.clientSecret
+
+    const result = await this.fetch(
+      'PUT',
+      `/auth/register/${this.oauthOptions.clientID}`,
+      data,
+      {
+        credentials: this.registrationAccessTokenToAuthHeader()
+      }
+    )
+
+    this.oauthOptions = this.camelCaseOAuthData(result)
+
+    return this.oauthOptions
+  }
+
+  /**
+   * Generates a random state code to be used during the OAuth process
+   * @returns {string}
+   */
+  generateStateCode() {
+    const STATE_SIZE = 16
+    const hasCrypto =
+      typeof window !== 'undefined' &&
+      typeof window.crypto !== 'undefined' &&
+      typeof window.crypto.getRandomValues === 'function'
+
+    let buffer
+    if (hasCrypto) {
+      buffer = new Uint8Array(STATE_SIZE)
+      window.crypto.getRandomValues(buffer)
+    } else {
+      buffer = new Array(STATE_SIZE)
+      for (let i = 0; i < buffer.length; i++) {
+        buffer[i] = Math.floor(Math.random() * 255)
+      }
+    }
+
+    return btoa(String.fromCharCode.apply(null, buffer))
+      .replace(/=+$/, '')
+      .replace(/\//g, '_')
+      .replace(/\+/g, '-')
+  }
```

Comment:
It's not very easy to see what this code do. Is it ok if it returns `0123456789abcdef`? What is a sample? Is it tested somewhere?

Reply:
Another confusion comes from the owner of this function: how is it related to this class? It may be written in its own file or in a private area of the file of the class.

Reply:
> Another confusion comes from the owner of this function: how is it related to this class? It may be written in its own file or in a private area of the file of the class.

You're right, it could be a standalone function. You would always use it in conjunction with the rest of the class though, so you'd have to import it along with the class everywhere. 
I don't mind moving it, but it would make the DX worse. Do you think it's worth sacrificing a bit of DX for the clarity it will give us?

Otherwise, it just… generates a random string. `0123456789abcdef` is cool. It has a test attached to it, and it's documented. Do you think a sample return value in the doc would help?

Reply:
If you don't need to export it, you could put it outside the class in the same file.

Yes, an example of returned valued may help.